pandas/core/frame.py

```diff
@@ -5444,6 +5444,34 @@ def shift(
     ) -> DataFrame:
         axis = self._get_axis_number(axis)
 
+        # Handle the case of multiple shifts
+        if is_list_like(periods):
+
+            new_df = DataFrame()
+
+            from pandas.core.reshape.concat import concat
+
+            for i in periods:
+                if not isinstance(i, int):
```

Comment:
Check all at once

Reply:
Is this in reference to the isinstance call? If I pass the whole list for shifts into the function and am checking for ints, then it would return false.

Reply:
Yes, and the error. If a List of strings is passed you would raise for the first. User fixes first, you raise for the second, etc. the error should display all invalid values at once

Reply:
So we can find all the indices that don't contain ints, and then raise the typerror.  i.e. check if each is an int in a separate loop above, and indicate all indices that aren't ints to the user

Reply:
Yep